packages/cli/src/install/command.ts

```diff
@@ -0,0 +1,71 @@
+import BrownieDetector from "../core/detection/brownie";
+import CRADetector from "../core/detection/cra";
+import FoundryDetector from "../core/detection/foundry";
+import HardhatDetector from "../core/detection/hardhat";
+import NextDetector from "../core/detection/next";
+import NPMDetector from "../core/detection/npm";
+import TruffleDetector from "../core/detection/truffle";
+import ViteDetector from "../core/detection/vite";
+import YarnDetector from "../core/detection/yarn";
+import { runCommand } from "../create/helpers/run-command";
+
+export async function install(distPath = "dist", projectPath = ".") {
+  const supportedContractFrameworks = [
+    new BrownieDetector(),
+    new FoundryDetector(),
+    new HardhatDetector(),
+    new TruffleDetector(),
+  ].filter((detector) => detector.matches(projectPath));
+
+  const supportedAppFrameworks = [
+    new ViteDetector(),
+    new NextDetector(),
+    new CRADetector(),
+  ].filter((detector) => detector.matches(projectPath));
+
+  const hasYarn = new YarnDetector().matches(".");
+  const hasNPM = new NPMDetector().matches(".");
+
+  const possibleProjects = [
+    ...supportedContractFrameworks,
+    ...supportedAppFrameworks,
+  ];
+
+  if (possibleProjects.length === 0) {
+    throw new Error("No supported project detected");
+  }
+
+  const dependencies = new Set<string>();
+
+  supportedContractFrameworks.forEach((detector) => {
+    console.log(`Detected ${detector.projectType} project`);
+
+    dependencies.add("@thirdweb-dev/contract");
+  });
+
+  supportedAppFrameworks.forEach((detector) => {
+    console.log(`Detected ${detector.projectType} project`);
+
+    dependencies.add("@thirdweb-dev/react");
+    dependencies.add("@thirdweb-dev/sdk");
+    dependencies.add("ethers^5");
```

Comment:
we should only do this if it's not already part of the dependencies of the project

Reply:
if you do @5 it will always install the latest 5 even if they happened to be on some lower 5 before (and we should not touch that)